### PR TITLE
124 send preferred translation

### PR DESCRIPTION
**Pull request by DanChirtoaca**
#168 and #169 should also be addressed

### CODE REVIEW

File: src/umr/static/scripts/app/subscription/FeedSubscriber.js [hunk 1]
```diff
@@ -13,7 +13,9 @@ export default class FeedSubscriber {
      */
     constructor(subscriptionList) {
         this.subscriptionList = subscriptionList;
-        this.currentLanguage = 'nl';
+        this.currentLanguage = 'nl'; // default 
+        var callback = (data) => this._setCurrentLanguage(data);
```
Comment (Lukeslux) [inline src/umr/static/scripts/app/subscription/FeedSubscriber.js]: This one is a bit redundant right? We do not change the method in any way.

File: src/umr/static/scripts/app/translation/Translator.js [hunk 2]
```diff
@@ -53,6 +53,23 @@ export default class Translator {
...
+     * @param {jQuery} $zeeguu - Zeeguu reference tag for which to send the user suggestion.
+     */
+    sendSuggestion ($zeeguu) {
+        var word = $zeeguu.children(config.HTML_ORIGINAL).text();
+        var context = this._getContext($zeeguu.get(0));
+        var url = $(config.HTML_ID_ARTICLE_URL).find('a').attr('href');
+        var title = $(config.HTML_ID_ARTICLE_TITLE).text();
+        var translation = $zeeguu.children(config.HTML_TRANSLATED).attr(config.HTML_ATTRIBUTE_SUGGESTION);
+
+        var callback = (data) => console.log(data);
```
Comment (Lukeslux) [inline src/umr/static/scripts/app/translation/Translator.js]: similar remark

File: src/umr/static/scripts/app/translation/Translator.js [hunk 3]
```diff
@@ -53,6 +53,23 @@ export default class Translator {
...
+        // Launch Zeeguu request to supply translation suggestion.
+        ZeeguuRequests.post(config.POST_TRANSLATION_SUGGESTION + '/' + FROM_LANGUAGE + '/' + config.TO_LANGUAGE,
```
Comment (Lukeslux) [inline src/umr/static/scripts/app/translation/Translator.js]: Just a note: We should probably find a different way to determine FROM_LANGUAGE than to let flask generate this (javascript can retrieve it from the url itself). But that's not related to this pull.

File: src/umr/static/scripts/app/translation/Translator.js [hunk 4]
```diff
@@ -53,6 +53,23 @@ export default class Translator {
     }
 
     /**
+     * Sends a post request to Zeeguu about a contribution/suggestion for a translation from user.
+     * @param {jQuery} $zeeguu - Zeeguu reference tag for which to send the user suggestion.
+     */
+    sendSuggestion ($zeeguu) {
+        var word = $zeeguu.children(config.HTML_ORIGINAL).text();
+        var context = this._getContext($zeeguu.get(0));
```
Comment (Lukeslux) [inline src/umr/static/scripts/app/translation/Translator.js]: why not just $zeeguu?

File: src/umr/static/scripts/app/subscription/FeedSubscriber.js [hunk 5]
```diff
@@ -46,6 +48,14 @@ export default class FeedSubscriber {
     }
 
     /**
+     * Sets the current language that user is studying (based on zeeguu settings).
+     * @param {string} langauge - The language code.
+     */
+    _setCurrentLanguage(language) {
+        this.currentLanguage = language;
```
Comment (Lukeslux) [inline src/umr/static/scripts/app/subscription/FeedSubscriber.js]: is `this` truly `this` here?

File: src/umr/static/scripts/app/translation/Translator.js (hunk 4 above)
Comment (DanChirtoaca) [inline src/umr/static/scripts/app/translation/Translator.js]: _getContext() expects DOM element not jQuery object

File: src/umr/static/scripts/app/translation/Translator.js (hunk 2 above)
Comment (DanChirtoaca) [inline src/umr/static/scripts/app/translation/Translator.js]: this one is indeed redundant, used to check what zeeguu will return, will remove it

**Pull request merged**